include/deal.II/base/index_set.h

```diff
@@ -256,6 +256,14 @@ class IndexSet
   size_type index_within_set (const size_type global_index) const;
 
   /**
+   * Same as above, but returns <code>true</code> if the global index is
+   * within the set and <code>false</code> otherwise. In the former case
+   * the how-manyth element of this set is set in @p local index.
+   */
+  bool index_within_set (const size_type global_index,
+                         size_type &local_index) const;
```

Comment:
> One could either return an integer that happens to be numbers::unsigned_int if the index is not in the set, or a regular size_type otherwise

I like this one, will update soon (double checking compilation which takes forever...)